### PR TITLE
chore: add publish-github-release action

### PR DESCRIPTION
**Pull request by chunshao90**
## Rationale
Add automatic publish github release workflow.

## Detailed Changes
Supports automatic generation of release content and divides pr into the following categories:
* Breaking Changes
* Features
* Refactor
* Fixed
* Docs
* Chore

## Test Plan
Manual test.

### CODE REVIEW

File: .github/release.yml
Comment (waynexia) [inline .github/release.yml:1]: Where would this file be used?

**Comment by chunshao90** [comment]
I have applied for infra to `add softprops/action-gh-release@v2` to the allow list for execution.
https://issues.apache.org/jira/browse/INFRA-26208?filter=-2

File: .github/release.yml
Comment (jiacai2050) [reply .github/release.yml:1]: This config file is used by GitHub for generate release note.

https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes#example-configurations

File: .github/release.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    - title: Refactor
+      labels:
+        - refactor
+    - title: Fixed
+      labels:
+        - bug
+    - title: Docs
+      labels:
+        - docs
+    - title: Chore
```
Comment (jiacai2050) [inline .github/release.yml:35]: Please add test, build, style.
- See https://github.com/apache/horaedb/pull/1581

Comment (chunshao90) [reply .github/release.yml:35]: done

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**